## Carregando dados brutos

In [1]:
df_2023_01 = pd.read_excel("data/SPDadosCriminais_2023.xlsx", sheet_name='JAN_JUN_2023')
df_2023_02 = pd.read_excel("data/SPDadosCriminais_2023.xlsx", sheet_name='JUL_DEZ_2023')
df_2024_01 = pd.read_excel("data/SPDadosCriminais_2024.xlsx", sheet_name='JAN-JUN_2024')
df_2024_02 = pd.read_excel("data/SPDadosCriminais_2024.xlsx", sheet_name='JUL-DEZ_2024')

NameError: name 'pd' is not defined

In [ ]:
df_2024_01.rename(columns={
                            'NOME_MUNICIPIO': 'CIDADE', ### 
                            'DATA_REGISTRO': 'DATA_COMUNICACAO', ###
                            'DESCR_SUBTIPOLOCAL': 'DESCR_TIPOLOCAL', ###
                             }, inplace=True)

df_2024_02.rename(columns={
                            'NOME_MUNICIPIO': 'CIDADE', ### 
                            'DATA_REGISTRO': 'DATA_COMUNICACAO', ###
                            'DESCR_SUBTIPOLOCAL': 'DESCR_TIPOLOCAL', ###
                             }, inplace=True)

In [ ]:
df_2024_02.shape

(599423, 25)

In [ ]:
df_2023_01.columns

Index(['NOME_DEPARTAMENTO', 'NOME_SECCIONAL', 'NOME_DELEGACIA', 'CIDADE',
       'NUM_BO', 'ANO_BO', 'DATA_COMUNICACAO', 'DATA_OCORRENCIA_BO',
       'HORA_OCORRENCIA_BO', 'DESC_PERIODO', 'DESCR_TIPOLOCAL', 'BAIRRO',
       'LOGRADOURO', 'NUMERO_LOGRADOURO', 'LATITUDE', 'LONGITUDE',
       'NOME_DELEGACIA_CIRCUNSCRIÇÃO', 'NOME_DEPARTAMENTO_CIRCUNSCRIÇÃO',
       'NOME_SECCIONAL_CIRCUNSCRIÇÃO', 'NOME_MUNICIPIO_CIRCUNSCRIÇÃO',
       'RUBRICA', 'DESCR_CONDUTA', 'NATUREZA_APURADA', 'MES_ESTATISTICA',
       'ANO_ESTATISTICA'],
      dtype='object')

In [ ]:
df_2024_01.columns

Index(['NOME_DEPARTAMENTO', 'NOME_SECCIONAL', 'NOME_DELEGACIA', 'CIDADE',
       'ANO_BO', 'NUM_BO', 'DATA_COMUNICACAO', 'DATA_OCORRENCIA_BO',
       'HORA_OCORRENCIA_BO', 'DESC_PERIODO', 'DESCR_TIPOLOCAL', 'BAIRRO',
       'LOGRADOURO', 'NUMERO_LOGRADOURO', 'LATITUDE', 'LONGITUDE',
       'NOME_DELEGACIA_CIRCUNSCRIÇÃO', 'NOME_DEPARTAMENTO_CIRCUNSCRIÇÃO',
       'NOME_SECCIONAL_CIRCUNSCRIÇÃO', 'NOME_MUNICIPIO_CIRCUNSCRIÇÃO',
       'RUBRICA', 'DESCR_CONDUTA', 'NATUREZA_APURADA', 'MES_ESTATISTICA',
       'ANO_ESTATISTICA'],
      dtype='object')

In [ ]:
dataframes = [df_2023_01, df_2023_02, df_2024_01, df_2024_02]
dados_criminais = pd.concat(dataframes)

In [ ]:
dados_criminais.shape

(2433725, 25)

In [ ]:
#descarregando o csv para facilitar
dados_criminais.to_csv("data/dados_criminais_merge.csv", index=False)

# Limpeza e organização da base

In [10]:
import pandas as pd
import numpy as np 

pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import warnings
warnings.filterwarnings("ignore")

In [11]:
dados = pd.read_csv('data/dados_criminais_merge.csv')

In [12]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2433725 entries, 0 to 2433724
Data columns (total 25 columns):
 #   Column                           Dtype  
---  ------                           -----  
 0   NOME_DEPARTAMENTO                object 
 1   NOME_SECCIONAL                   object 
 2   NOME_DELEGACIA                   object 
 3   CIDADE                           object 
 4   NUM_BO                           object 
 5   ANO_BO                           int64  
 6   DATA_COMUNICACAO                 object 
 7   DATA_OCORRENCIA_BO               object 
 8   HORA_OCORRENCIA_BO               object 
 9   DESC_PERIODO                     object 
 10  DESCR_TIPOLOCAL                  object 
 11  BAIRRO                           object 
 12  LOGRADOURO                       object 
 13  NUMERO_LOGRADOURO                object 
 14  LATITUDE                         object 
 15  LONGITUDE                        float64
 16  NOME_DELEGACIA_CIRCUNSCRIÇÃO     object 
 17  NOME_DEP

In [13]:
dados.head(3)

NOME_DEPARTAMENTO        NOME_SECCIONAL        NOME_DELEGACIA  \
0                          DECAP     DEL.SEC.1º CENTRO        01ª DDM CENTRO   
1                          DECAP     DEL.SEC.1º CENTRO        01ª DDM CENTRO   
2  DIPOL - DEPTO DE INTELIGENCIA  DELEGACIA ELETRONICA  DELEGACIA ELETRONICA   

    CIDADE  NUM_BO  ANO_BO DATA_COMUNICACAO DATA_OCORRENCIA_BO  \
0  S.PAULO  AP5096    2023       2023-01-14         2023-01-12   
1  S.PAULO  BD2233    2023       2023-01-25         2023-01-24   
2  S.PAULO  AA0063    2023       2022-12-31         2022-12-30   

  HORA_OCORRENCIA_BO DESC_PERIODO DESCR_TIPOLOCAL BAIRRO  \
0           21:00:19          NaN           Praça     SE   
1           20:40:00          NaN     Apartamento    LUZ   
2                NaN      A noite     Via Pública     SÉ   

                                  LOGRADOURO NUMERO_LOGRADOURO LATITUDE  \
0  VEDAÇÃO DA DIVULGAÇÃO DOS DADOS RELATIVOS                 0      0.0   
1  VEDAÇÃO DA DIVULGAÇÃO DOS DADOS RELATIVOS                 0      0.0   
2                                PRAÇA DA SÉ                 0      NaN   

   LONGITUDE NOME_DELEGACIA_CIRCUNSCRIÇÃO NOME_DEPARTAMENTO_CIRCUNSCRIÇÃO  \
0        0.0                  01º D.P. SE                           DECAP   
1        0.0                  01º D.P. SE                           DECAP   
2        NaN                  01º D.P. SE                           DECAP   

  NOME_SECCIONAL_CIRCUNSCRIÇÃO NOME_MUNICIPIO_CIRCUNSCRIÇÃO  \
0            DEL.SEC.1º CENTRO                      S.PAULO   
1            DEL.SEC.1º CENTRO                      S.PAULO   
2            DEL.SEC.1º CENTRO                      S.PAULO   

              RUBRICA DESCR_CONDUTA NATUREZA_APURADA  MES_ESTATISTICA  \
0  Art. 213 - Estupro           NaN          ESTUPRO                1   
1  Art. 213 - Estupro           NaN          ESTUPRO                1   
2    Furto (art. 155)        Outros   FURTO - OUTROS                1   

   ANO_ESTATISTICA  
0             2023  
1             2023  
2             2023

### Filtrando dados da cidade de São Paulo

In [14]:
dados = dados.loc[dados['CIDADE'] == 'S.PAULO']
dados['CIDADE'] = dados['CIDADE'].str.replace('S.PAULO', 'São Paulo')

In [15]:
dados['CIDADE'].unique()

array(['São Paulo'], dtype=object)

### Transformando os dtypes - Data, Hora, Criando novas colunas qualitativas

In [16]:
dados['NUMERO_LOGRADOURO'] = pd.to_numeric(dados['NUMERO_LOGRADOURO'],errors='coerce')
dados['NUMERO_LOGRADOURO'] = dados['NUMERO_LOGRADOURO'].fillna(0).astype(int)

In [17]:
dados['LATITUDE'] = pd.to_numeric(dados['LATITUDE'],errors='coerce')

In [18]:
dados['DATA_COMUNICACAO'] = pd.to_datetime(dados['DATA_COMUNICACAO'],
                                          errors='coerce', format = '%Y-%m-%d')
dados['DATA_OCORRENCIA_BO'] = pd.to_datetime(dados['DATA_OCORRENCIA_BO'],
                                          errors='coerce', format = '%Y-%m-%d')

In [19]:
dados[['DATA_COMUNICACAO', 'DATA_OCORRENCIA_BO']].info()

<class 'pandas.core.frame.DataFrame'>
Index: 978411 entries, 0 to 2433633
Data columns (total 2 columns):
 #   Column              Non-Null Count   Dtype         
---  ------              --------------   -----         
 0   DATA_COMUNICACAO    978411 non-null  datetime64[ns]
 1   DATA_OCORRENCIA_BO  977884 non-null  datetime64[ns]
dtypes: datetime64[ns](2)
memory usage: 22.4 MB


In [20]:
dados['DIA_DA_SEMANA'] = dados['DATA_OCORRENCIA_BO'].dt.day_name()
dados['MÊS'] = dados['DATA_OCORRENCIA_BO'].dt.month_name()

In [21]:
dias_semana = {
    "Monday": "Segunda-feira",
    "Tuesday": "Terça-feira",
    "Wednesday": "Quarta-feira",
    "Thursday": "Quinta-feira",
    "Friday": "Sexta-feira",
    "Saturday": "Sábado",
    "Sunday": "Domingo"
}

dados['DIA_DA_SEMANA'] = dados['DIA_DA_SEMANA'].map(dias_semana)

In [22]:
mes = {
       'January':'Janeiro',
       'December':'Dezembro', 
       'October':'Outubro', 
       'November':'Novembro', 
       'September':'Setembro',
       'August':'Agosto', 
       'May':'Maio', 
       'June':'Junho', 
       'July':'Julho', 
       'April':'Abril', 
       'March':'Março', 
       'February':'Fevereiro'
}

dados['MÊS'] = dados['MÊS'].map(mes)

In [23]:
dados.head(3)

NOME_DEPARTAMENTO        NOME_SECCIONAL        NOME_DELEGACIA  \
0                          DECAP     DEL.SEC.1º CENTRO        01ª DDM CENTRO   
1                          DECAP     DEL.SEC.1º CENTRO        01ª DDM CENTRO   
2  DIPOL - DEPTO DE INTELIGENCIA  DELEGACIA ELETRONICA  DELEGACIA ELETRONICA   

      CIDADE  NUM_BO  ANO_BO DATA_COMUNICACAO DATA_OCORRENCIA_BO  \
0  São Paulo  AP5096    2023       2023-01-14         2023-01-12   
1  São Paulo  BD2233    2023       2023-01-25         2023-01-24   
2  São Paulo  AA0063    2023       2022-12-31         2022-12-30   

  HORA_OCORRENCIA_BO DESC_PERIODO DESCR_TIPOLOCAL BAIRRO  \
0           21:00:19          NaN           Praça     SE   
1           20:40:00          NaN     Apartamento    LUZ   
2                NaN      A noite     Via Pública     SÉ   

                                  LOGRADOURO  NUMERO_LOGRADOURO  LATITUDE  \
0  VEDAÇÃO DA DIVULGAÇÃO DOS DADOS RELATIVOS                  0       0.0   
1  VEDAÇÃO DA DIVULGAÇÃO DOS DADOS RELATIVOS                  0       0.0   
2                                PRAÇA DA SÉ                  0       NaN   

   LONGITUDE NOME_DELEGACIA_CIRCUNSCRIÇÃO NOME_DEPARTAMENTO_CIRCUNSCRIÇÃO  \
0        0.0                  01º D.P. SE                           DECAP   
1        0.0                  01º D.P. SE                           DECAP   
2        NaN                  01º D.P. SE                           DECAP   

  NOME_SECCIONAL_CIRCUNSCRIÇÃO NOME_MUNICIPIO_CIRCUNSCRIÇÃO  \
0            DEL.SEC.1º CENTRO                      S.PAULO   
1            DEL.SEC.1º CENTRO                      S.PAULO   
2            DEL.SEC.1º CENTRO                      S.PAULO   

              RUBRICA DESCR_CONDUTA NATUREZA_APURADA  MES_ESTATISTICA  \
0  Art. 213 - Estupro           NaN          ESTUPRO                1   
1  Art. 213 - Estupro           NaN          ESTUPRO                1   
2    Furto (art. 155)        Outros   FURTO - OUTROS                1   

   ANO_ESTATISTICA DIA_DA_SEMANA       MÊS  
0             2023  Quinta-feira   Janeiro  
1             2023   Terça-feira   Janeiro  
2             2023   Sexta-feira  Dezembro

In [24]:
dados['DESC_PERIODO'].unique()

array([nan, 'A noite', 'Pela manhã', 'A tarde', 'De madrugada',
       'Em hora incerta', 'PELA MANHÃ', 'EM HORA INCERTA'], dtype=object)

In [25]:
dados['DESC_PERIODO'] = dados['DESC_PERIODO'].str.strip()
dados['DESC_PERIODO'] = dados['DESC_PERIODO'].str.capitalize()

In [26]:
dados['DESC_PERIODO'].unique()

array([nan, 'A noite', 'Pela manhã', 'A tarde', 'De madrugada',
       'Em hora incerta'], dtype=object)

In [27]:
dados['DESC_PERIODO'].isna().sum()

647002

A descrição do período da ocorrência é algo muito importante para a análise mas temos 647002 dados nulos nessa coluna.  
Para melhorar a contagem de nulos podemos extrair esses valores da coluna de Hora da Ocorrência. 

In [28]:
# transformando a coluna em datetime
dados['HORA_OCORRENCIA_BO'] = pd.to_datetime(dados['HORA_OCORRENCIA_BO'],
                                             format='%H:%M:%S', errors='coerce')

# extraindo apenas a hora da ocorrência
dados['HORA_OCORRENCIA_BO'] = dados['HORA_OCORRENCIA_BO'].dt.hour

In [29]:
# função simples para a atribuição do períoco da ocorrência

def descricao_periodo(hour):
    if 0 <= hour < 6:
        return 'De madrugada'
    elif 6 <= hour < 12:
        return 'Pela manhã'
    elif 12 <= hour < 18:
        return 'A tarde'
    elif 18 <= hour <= 23:
        return 'A noite'
    else:
        return 'Em hora incerta'

In [30]:
# aplicando a função em uma nova coluna
dados['PERIODO'] = dados['HORA_OCORRENCIA_BO'].apply(descricao_periodo)

In [31]:
# comparando as colunas e conferindo os valores
dados[['HORA_OCORRENCIA_BO','DESC_PERIODO','PERIODO']].head(20)

HORA_OCORRENCIA_BO  DESC_PERIODO          PERIODO
0                 21.0           NaN          A noite
1                 20.0           NaN          A noite
2                  NaN       A noite  Em hora incerta
3                  NaN    Pela manhã  Em hora incerta
4                  NaN       A tarde  Em hora incerta
5                 21.0           NaN          A noite
6                  NaN  De madrugada  Em hora incerta
7                  NaN       A tarde  Em hora incerta
8                  7.0           NaN       Pela manhã
9                  NaN    Pela manhã  Em hora incerta
10                16.0           NaN          A tarde
11                20.0           NaN          A noite
12                20.0           NaN          A noite
13                 NaN       A tarde  Em hora incerta
14                 NaN       A noite  Em hora incerta
15                 NaN    Pela manhã  Em hora incerta
16                 NaN       A tarde  Em hora incerta
17                14.0           NaN          A tarde
18                 NaN    Pela manhã  Em hora incerta
19                 NaN    Pela manhã  Em hora incerta

In [32]:
# preenchendo os valores nulos da coluna original com valores extraídos pela função 
dados['DESC_PERIODO'] = dados['DESC_PERIODO'].fillna(dados['PERIODO'])

In [33]:
# diminuímos consideravelmente o número de valores nulos! 
dados['DESC_PERIODO'].isna().sum()

0

In [34]:
dados['BAIRRO'].unique()

array(['SE', 'LUZ', 'SÉ', ..., 'JARDIM GUARULHOS', 'VL DOLORES',
       'ITARARÉ'], dtype=object)

A base tem muitos bairros com valores incorretos e diferenças de grafia, esse problema atrapalha na agregação das ocorrencias. Aqui decidi filtrar os 150 bairros com maiores ocorrencias para arrumar os valores divergentes.

In [35]:
dados['BAIRRO'] = dados['BAIRRO'].str.strip()
dados['BAIRRO'] = dados['BAIRRO'].str.title()

In [36]:
dados['BAIRRO'].nunique()

2961

In [37]:
dados['BAIRRO'].value_counts().head(150)

BAIRRO
Pinheiros                              22457
Bela Vista                             19708
Republica                              18979
Santo Amaro                            17151
Vila Mariana                           16456
Santana                                15958
Barra Funda                            15438
Itaim Bibi                             14971
Capao Redondo                          13856
Ipiranga                               13387
Grajau                                 13360
Bom Retiro                             13083
Itaquera                               13064
Se                                     12645
Liberdade                              12450
Jardim Paulista                        11931
Cidade Dutra                           11435
Consolacao                             10989
Campo Limpo                            10900
Tatuape                                10382
Jabaquara                              10352
Mooca                                  10035
Brá

In [38]:
bairros = {'Se': 'Sé',
           'Bras': 'Brás',
           'Agua branca': 'Água Branca',
           'Alianca': 'Aliança',
           'Americanopolis': 'Americanópolis',
           'Anhangabau': 'Anhangabaú',
           'Brasilandia': 'Brasilândia',
           'Broklin': 'Brooklin',
           'Republica': 'República',
           'Capao redondo': 'Capão Redondo',
           'Consolacao': 'Consolação',
           'Tatuape': 'Tatuapé',
           'Jd são luis': 'Jardim São Luís',
           'Sacoma': 'Sacomã',
           'Freguesia do o': 'Freguesia do Ó',
           'Saude': 'Saúde',
           'Tremembe': 'Tremembé',
           'Sao Mateus': 'São Mateus',
           'Jaragua': 'Jaraguá',
           'Vila Curuca': 'Vila Curuça',
           'Sao Lucas': 'São Lucas',
           'Centro Histórico De São Paulo': 'Centro',
           'Butanta': 'Butantã',
           'Belem': 'Belém',
           'Cidade Lider': 'Cidade Líder',
           'Agua Rasa': 'Água Rasa',
           'Limao': 'Limão',
           'Carrao': 'Vila Carrão',
           'Jacana': 'Jaçanã',
           'Sao Rafael': 'São Rafael',
           'Sao Domingos': 'São Domingos',
           'Jaguare': 'Jaguaré',
           'Pompeia': 'Pompéia',
           'Guaianazes': "Guaianases",
           "Santa Efigênia": "Santa Ifigênia",
           "Jd São Luis": "Jardim São Luís"}

dados['BAIRRO'] = dados['BAIRRO'].replace(bairros)

In [41]:
dados['NATUREZA_APURADA'].unique()

array(['ESTUPRO', 'FURTO - OUTROS', 'FURTO DE CARGA', 'FURTO DE VEÍCULO',
       'HOMICÍDIO CULPOSO POR ACIDENTE DE TRÂNSITO', 'HOMICÍDIO DOLOSO',
       'LESÃO CORPORAL CULPOSA - OUTRAS',
       'LESÃO CORPORAL CULPOSA POR ACIDENTE DE TRÂNSITO',
       'LESÃO CORPORAL DOLOSA', 'PORTE DE ARMA', 'PORTE DE ENTORPECENTES',
       'ROUBO - OUTROS', 'ROUBO DE VEÍCULO', 'TRÁFICO DE ENTORPECENTES',
       'APREENSÃO DE ENTORPECENTES', 'ROUBO DE CARGA',
       'ESTUPRO DE VULNERÁVEL', 'LATROCÍNIO', 'TENTATIVA DE HOMICÍDIO',
       'LESÃO CORPORAL SEGUIDA DE MORTE', 'EXTORSÃO MEDIANTE SEQUESTRO',
       'HOMICÍDIO CULPOSO OUTROS',
       'HOMICÍDIO DOLOSO POR ACIDENTE DE TRÂNSITO', 'ROUBO A BANCO',
       'Furto de Veículo', 'Roubo de Veículo'], dtype=object)

In [42]:
dados['NATUREZA_APURADA'] = dados['NATUREZA_APURADA'].str.strip()
dados['NATUREZA_APURADA'] = dados['NATUREZA_APURADA'].str.capitalize()

In [43]:
dados['NATUREZA_APURADA'].unique()

array(['Estupro', 'Furto - outros', 'Furto de carga', 'Furto de veículo',
       'Homicídio culposo por acidente de trânsito', 'Homicídio doloso',
       'Lesão corporal culposa - outras',
       'Lesão corporal culposa por acidente de trânsito',
       'Lesão corporal dolosa', 'Porte de arma', 'Porte de entorpecentes',
       'Roubo - outros', 'Roubo de veículo', 'Tráfico de entorpecentes',
       'Apreensão de entorpecentes', 'Roubo de carga',
       'Estupro de vulnerável', 'Latrocínio', 'Tentativa de homicídio',
       'Lesão corporal seguida de morte', 'Extorsão mediante sequestro',
       'Homicídio culposo outros',
       'Homicídio doloso por acidente de trânsito', 'Roubo a banco'],
      dtype=object)

In [44]:
dados['DESCR_TIPOLOCAL'].value_counts().head(10)

DESCR_TIPOLOCAL
Via Pública                                736952
Metroviário e Ferroviário Metropolitano     53763
Outros                                      41488
Casa                                        35222
Terminal/Estação                             5534
Residência                                   4842
Transeunte                                   4537
Apartamento                                  4070
Mercado                                      3670
Interior de Transporte Coletivo              3656
Name: count, dtype: int64

Agora vamos fazer transformações pensando no nosso Dashboard.  
O Tableau exige uma certa hierarquia para fazer visualizações geográficas então agora vamos criar essas colunas!

In [45]:
dados['PAÍS'] = 'Brasil'
dados['ESTADO'] = 'São Paulo'

In [46]:
dados.rename(columns={'LATITUDE': 'LAT',
                      'LONGITUDE': 'LONG',
                      'ANO_ESTATISTICA': 'ANO'}, inplace=True)

In [47]:
dados.columns

Index(['NOME_DEPARTAMENTO', 'NOME_SECCIONAL', 'NOME_DELEGACIA', 'CIDADE',
       'NUM_BO', 'ANO_BO', 'DATA_COMUNICACAO', 'DATA_OCORRENCIA_BO',
       'HORA_OCORRENCIA_BO', 'DESC_PERIODO', 'DESCR_TIPOLOCAL', 'BAIRRO',
       'LOGRADOURO', 'NUMERO_LOGRADOURO', 'LAT', 'LONG',
       'NOME_DELEGACIA_CIRCUNSCRIÇÃO', 'NOME_DEPARTAMENTO_CIRCUNSCRIÇÃO',
       'NOME_SECCIONAL_CIRCUNSCRIÇÃO', 'NOME_MUNICIPIO_CIRCUNSCRIÇÃO',
       'RUBRICA', 'DESCR_CONDUTA', 'NATUREZA_APURADA', 'MES_ESTATISTICA',
       'ANO', 'DIA_DA_SEMANA', 'MÊS', 'PERIODO', 'PAÍS', 'ESTADO'],
      dtype='object')

In [ ]:
# filtrando apenas os atributos que iremos utilizar no dashboard do tableau
dados = dados[['NOME_DEPARTAMENTO', 'NOME_SECCIONAL', 'NOME_DELEGACIA', 'NATUREZA_APURADA', 
                       'NUM_BO', 'ANO_BO', 'DATA_OCORRENCIA_BO', 'HORA_OCORRENCIA_BO', 
                       'ANO', 'DIA_DA_SEMANA', 'MÊS', 'DESC_PERIODO', 'DESCR_TIPOLOCAL', 
                       'PAÍS', 'ESTADO', 'CIDADE', 'LAT', 'LONG', 'BAIRRO', 'LOGRADOURO', 'NUMERO_LOGRADOURO']]

In [49]:
dados.head(3)

NOME_DEPARTAMENTO        NOME_SECCIONAL        NOME_DELEGACIA  \
0                          DECAP     DEL.SEC.1º CENTRO        01ª DDM CENTRO   
1                          DECAP     DEL.SEC.1º CENTRO        01ª DDM CENTRO   
2  DIPOL - DEPTO DE INTELIGENCIA  DELEGACIA ELETRONICA  DELEGACIA ELETRONICA   

  NATUREZA_APURADA  NUM_BO  ANO_BO DATA_OCORRENCIA_BO  HORA_OCORRENCIA_BO  \
0          Estupro  AP5096    2023         2023-01-12                21.0   
1          Estupro  BD2233    2023         2023-01-24                20.0   
2   Furto - outros  AA0063    2023         2022-12-30                 NaN   

    ANO DIA_DA_SEMANA       MÊS DESC_PERIODO DESCR_TIPOLOCAL    PAÍS  \
0  2023  Quinta-feira   Janeiro      A noite           Praça  Brasil   
1  2023   Terça-feira   Janeiro      A noite     Apartamento  Brasil   
2  2023   Sexta-feira  Dezembro      A noite     Via Pública  Brasil   

      ESTADO     CIDADE  LAT  LONG BAIRRO  \
0  São Paulo  São Paulo  0.0   0.0     Sé   
1  São Paulo  São Paulo  0.0   0.0    Luz   
2  São Paulo  São Paulo  NaN   NaN     Sé   

                                  LOGRADOURO  NUMERO_LOGRADOURO  
0  VEDAÇÃO DA DIVULGAÇÃO DOS DADOS RELATIVOS                  0  
1  VEDAÇÃO DA DIVULGAÇÃO DOS DADOS RELATIVOS                  0  
2                                PRAÇA DA SÉ                  0

In [50]:
dados.to_csv("data/dados_tableau.csv", index=False)